scanners/ios/icon.go

```diff
@@ -0,0 +1,121 @@
+package ios
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bitrise-io/bitrise-init/models"
+	"github.com/bitrise-io/bitrise-init/utility"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/xcode-project/xcodeproj"
+)
+
+// lookupIconBySchemeName returns possible ios app icons for a scheme.
+func lookupIconBySchemeName(projectPath string, schemeName string, basepath string) (models.Icons, error) {
+	project, err := xcodeproj.Open(projectPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open project file: %s, error: %s", projectPath, err)
+	}
+
+	scheme, found := project.Scheme(schemeName)
+	if !found {
+		return nil, fmt.Errorf("scheme (%s) not found in project", schemeName)
+	}
+
+	mainTarget, err := mainTargetOfScheme(project, scheme.Name)
+
+	return lookupIconByTarget(projectPath, mainTarget, basepath)
+}
+
+// lookupIconByTargetName returns possible ios app icons for a target.
+func lookupIconByTargetName(projectPath string, targetName string, basepath string) (models.Icons, error) {
+	target, err := nameToTarget(projectPath, targetName)
+	if err != nil {
+		return models.Icons{}, nil
+	}
+
+	return lookupIconByTarget(projectPath, target, basepath)
+}
+
+func nameToTarget(projectPath string, targetName string) (xcodeproj.Target, error) {
+	project, err := xcodeproj.Open(projectPath)
```

Comment:
xcodeproj.Project stores the project path, so pass the project model instead of the path and open it again.